core/chains/evm/gas/block_history_estimator.go

```diff
@@ -299,7 +299,7 @@ func (b *BlockHistoryEstimator) checkConnectivity(attempts []EvmPriorAttempt) er
 		if attempt.GetBroadcastBeforeBlockNum() == nil {
 			// this shouldn't happen; any broadcast attempt ought to have a
 			// BroadcastBeforeBlockNum otherwise its an assumption violation
-			return errors.Errorf("BroadcastBeforeBlockNum was unexpectedly nil for attempt %s", attempt.GetHash().Hex())
+			return errors.Errorf("BroadcastBeforeBlockNum was unexpectedly nil for attempt %s", attempt.GetHash().String())
```

Comment:
The `String()` method should be called automatically when using the `%s` verb:
```suggestion
			return errors.Errorf("BroadcastBeforeBlockNum was unexpectedly nil for attempt %s", attempt.GetHash())
```
We could also consider implementing it on `attempt`:
```suggestion
			return errors.Errorf("BroadcastBeforeBlockNum was unexpectedly nil for attempt %s", attempt)
```

Reply:
thanks, done